### ISSUE TITLE
Weighted and macro are different

### ISSUE THREAD

**Issue by tuhinjubcse** [pinkeshbadjatiya/twitter-hatespeech#7]
p += precision_score(y_test, y_pred, average='weighted')
        p1 += precision_score(y_test, y_pred, average='micro')
        r += recall_score(y_test, y_pred, average='weighted')
        r1 += recall_score(y_test, y_pred, average='micro')
        f1 += f1_score(y_test, y_pred, average='weighted')
        f11 += f1_score(y_test, y_pred, average='micro')

    print "macro results are"
    print "average precision is %f" %(p/NO_OF_FOLDS)
    print "average recall is %f" %(r/NO_OF_FOLDS)
    print "average f1 is %f" %(f1/NO_OF_FOLDS)

    print "micro results are"
    print "average precision is %f" %(p1/NO_OF_FOLDS)
    print "average recall is %f" %(r1/NO_OF_FOLDS)
    print "average f1 is %f" %(f11/NO_OF_FOLDS)

another bug you are adding weighted f1 and reporting them as macro , both are different FYI . Macro scores will definitely be lower than weighted . You mention in your paper weighted macro , I am not sure what that means

**Comment by shashankg7** [comment]
Weighted Macro metric: Calculate metrics for each label, and find their average, weighted by support (the number of true instances for each label). This alters ‘macro’ to account for label imbalance; it can result in an F-score that is not between precision and recall.

Since dataset was skewed w.r.t classes, we used this metric to take into account the label imbalance. 

Please read carefully before pointing out something in the paper. 

Thanks!

**Comment by tuhinjubcse** [comment]
i know the difference between them , precisely why i am saying that . considering the task is to classify abusive tweet , since the support for None class is a lot high the 'weighted' score is gonna be definitely high which looks good on paper  . Usually in classification tasks macro F1 is reported . there is either weighted F1 or macro F1 . Sorry but there is no weighted macro .

**Comment by shashankg7** [comment]
http://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html#sklearn.metrics.f1_score
. Please look into it.

Thanks!
